src/xcode/ENA/ENA/Source/Models/ExposureWindowsMetadata/ExposureWindowsMetadataService.swift

```diff
@@ -0,0 +1,73 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+class ExposureWindowsMetadataService {
+	
+	// MARK: - Internal
+
+	func collectExposureWindows(from riskCalculation: RiskCalculationProtocol, store: Store) {
+		guard let calculation = riskCalculation as? RiskCalculation else {
+			Log.debug("Instance of riskCalculation couldn't be casted into RiskCalculation type", log: .ppa)
+			return
+		}
+		self.clearReportedExposureWindowsQueueIfNeeded(store: store)
+		
+		let mappedSubmittionExposureWindows: [SubmittionExposureWindow] = calculation.mappedExposureWindows.map {
+			SubmittionExposureWindow(
+				exposureWindow: $0.exposureWindow,
+				transmissionRiskLevel: $0.transmissionRiskLevel,
+				normalizedTime: $0.normalizedTime,
+				hash: self.convertToData($0.exposureWindow)?.sha256String(),
+				date: $0.date
+			)
+		}
+		
+		if let metadata = store.exposureWindowsMetadata {
+			// if store is initialized:
+			// - Queue if new: if the hash of the Exposure Window not included in reportedExposureWindowsQueue, the Exposure Window is added to reportedExposureWindowsQueue.
+			for exposureWindow in mappedSubmittionExposureWindows {
+				if metadata.reportedExposureWindowsQueue.contains(where: { $0.hash == exposureWindow.hash }) {
+					store.exposureWindowsMetadata?.newExposureWindowsQueue.append(exposureWindow)
+					store.exposureWindowsMetadata?.reportedExposureWindowsQueue.append(exposureWindow)
+				}
+			}
+		} else {
+			// if store is not initialized:
+			// - Initialize and add all of the exposure windows to both "newExposureWindowsQueue" and "reportedExposureWindowsQueue" arrays
+			store.exposureWindowsMetadata = ExposureWindowsMetadata(
+				newExposureWindowsQueue: mappedSubmittionExposureWindows,
+				reportedExposureWindowsQueue: mappedSubmittionExposureWindows
+			)
+		}
+	}
+	
+	// MARK: - Private
+
+	private func clearReportedExposureWindowsQueueIfNeeded(store: Store) {
+		if let nonExpiredWindows = store.exposureWindowsMetadata?.reportedExposureWindowsQueue.filter({
+			guard let day = Calendar.current.dateComponents([.day], from: $0.date, to: Date()).day else {
+				Log.debug("Exposure Window is removed from reportedExposureWindowsQueue as the date componant is nil", log: .ppa)
+				return false
+			}
+			return day < 15
+		}) {
+			store.exposureWindowsMetadata?.reportedExposureWindowsQueue = nonExpiredWindows
+		}
+	}
+	
+	private func convertToData(_ window: ExposureWindow) -> Data? {
```

Comment:
If I get it right this method is only used to later create an sha256 String?
```suggestion
	private func convertToData(_ window: ExposureWindow) -> String? {
```
Better create the sha256 string complete here?